Angular/Angular guidelines and best practices/Assets & caching.md

```diff
@@ -0,0 +1,172 @@
+- In certain cases, the logo image will be present two times in the final bundle:
+  - `/dist/logo.d808b08ce47c5d0c53cd.png` and `/dist/assets/logo.png`
+- `assets/logo.png` + `<base href="/web/">` would work and it would load the image from `https://app.com/web/assets/logo.png` if SCSS transpiled, but it does not. If you set this URL manually in devtools, you will notice that it does indeed work and takes into account `base` `href` value.
+- The last two entries from the table do work, but require source code changes if `base` `href` is changed.
+
+Conclusion here is a bit different than for `<img>` and `fetch`/`XHR` paths. For SCSS URLs we recommend using `^assets/…` in order to avoid assets file duplication and make it work for different `base` `href` values without the need to modify URLs in the source files. Keep in mind that this solution could easily stop working if there are some breaking changes in Webpack or the Angular CLI. Handbook will be updated with a better solution once and if it is found.
+
+## Index file
+
+The entry-point into a Single Page Application (SPA), like Angular, is the `index.html` file. During the build process, the source `src/index.html` is taken and updated with links to the generated JS and CSS chunks and placed into `dist/[project-name]/index.html`.
+
+Fingerprinted links to JS and CSS chunks will be different for each build, and the `index.html` file will be different as well. Because of this, it is important that the `index.html` file is not cached permanently on the client and that the client requests cache revalidation. If the client has a stale `index.html` file, it will try loading stale JS and CSS chunks as well. Those might or might not be in its cache, and depending on this the client might get a completely outdated app or a broken app that is missing some chunks.
+
+**Rule #1**: Do not cache the built `index.html` file. Revalidate it each time!
+
+## JS and CSS chunks
+
+JS and CSS chunks that are linked to from the `index.html` file are fingerprinted and because of that they can be cached indefinitely. If we create a new application build, the `index.html` file will be re-validated and the client will load new chunks.
+
+**Rule #2**: Cache fingerprinted JS and CSS files with a long expiration time!
+
+## Static assets
+
+The tricky part with serving Angular applications is caching the static assets files. Some frameworks, depending on Webpack configuration and the way that the assets are used, fingerprint all the images and other static assets. In Angular, static assets like images and custom fonts are placed in the `src/assets/` directory. This directory gets copied without any modifications to the final application build directory - `dist/[project-name]/assets/`.
+
+The files from the `assets/` directory do not get fingerprinted. Because of this, it is important to set up the correct caching mechanism for serving these files or else the client might be using stale files. There are two solutions:
```

Comment:
```suggestion
Files from the `assets/` directory do not get fingerprinted. Because of this, it is important to set up the correct caching mechanism for serving these files or else the client might be using stale files. There are two solutions:
```